alf/algorithms/sac_algorithm.py

```diff
@@ -681,38 +696,42 @@ def train_step(self, exp: Experience, state: SacState):
             log_pi = sum(nest.flatten(log_pi))
 
         if self._prior_actor is not None:
-            prior_step = self._prior_actor.train_step(exp, ())
+            prior_step = self._prior_actor.train_step(inputs, ())
             log_prior = dist_utils.compute_log_probability(
                 prior_step.output, action)
             log_pi = log_pi - log_prior
 
         actor_state, actor_loss = self._actor_train_step(
-            exp, state.actor, action, critics, log_pi, action_distribution)
+            inputs, state.actor, action, critics, log_pi, action_distribution)
         critic_state, critic_info = self._critic_train_step(
-            exp, state.critic, action, log_pi, action_distribution)
+            inputs, state.critic, rollout_info, action, action_distribution)
         alpha_loss = self._alpha_train_step(log_pi)
 
         state = SacState(
             action=action_state, actor=actor_state, critic=critic_state)
         info = SacInfo(
+            reward=inputs.reward,
+            step_type=inputs.step_type,
+            discount=inputs.discount,
```

Comment:
Since the adding of these fields are the same for many instance algorithms, it would be good if they can be done automatically or with a helper function.

Reply:
I thought about that before. But decided to do this way to make the code clearer.